server/mqtt.go

```diff
@@ -2764,55 +2775,65 @@ func (as *mqttAccountSessionManager) transferUniqueSessStreamsToMuxed(log *Serve
 	retry = false
 }
 
-func (as *mqttAccountSessionManager) transferRetainedToPerKeySubjectStream(log *Server) bool {
+func (as *mqttAccountSessionManager) transferRetainedToPerKeySubjectStream(log *Server) error {
 	jsa := &as.jsa
-	var count, errors int
+	var processed int
+	var transferred int
 
+	const timeoutAfter = 1 * time.Second
```

Comment:
This is I think way too low to transfer possibly large amount of retained messages. Knowing that now the client connection will fail, we need to make sure that this does not lead to too many times application not able to connect, or worse never able to since and error in this function is considered "fatal". 
I am thinking now that this is maybe too much of a risk: being unable to connect any client because an error transferring messages. For instance with your other PR changing the JSON encoding, you could imagine a situation similar in the future and that would prevent the server with this code to ever accept a client connection.

Reply:
👍  to your overall comment - make 1-time best effort, proceed with the update, attempt to delete the possible one remaining message.

Reply:
I did not remove the timeout altogether, upped it to 5s, but maybe still not enough?

@derekcollison  suggested using an ordered consumer to load them quicker, if I understood correctly. Have not gotten to that yet.